spark/src/main/java/org/apache/iceberg/spark/source/IcebergBatchWriter.java

```diff
@@ -112,8 +115,8 @@
     this.targetFileSize = options.getLong("target-file-size-bytes", tableTargetFileSize);
   }
 
-  private FileFormat getFileFormat(Map<String, String> tableProperties, DataSourceOptions options) {
-    Optional<String> formatOption = options.get("write-format");
+  protected FileFormat getFileFormat(Map<String, String> tableProperties, Map<String, String> options) {
+    Optional<String> formatOption = Optional.ofNullable(options.get("write-format"));
```

Comment:
This shouldn't introduce Optional now that the API no longer passes it. Instead, let's do something like this:
```
options.getOrDefault("write-format",
    tableProperties.getOrDefault(DEFAULT_FILE_FORMAT, DEFAULT_FILE_FORMAT_DEFAULT))
```